onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -18,9 +18,12 @@ limitations under the License.
 
 // build\windows\debug\external\eigen3\unsupported\eigen\cxx11\src/Tensor/Tensor.h(76):
 // warning C4554: '&': check operator precedence for possible error; use parentheses to clarify precedence
+// build\windows\debug\external\eigen3\unsupported\eigen\cxx11\src/Tensor/TensorStorage.h(65):
+// warning C4324: structure was padded due to alignment specifier
 #ifdef _MSC_VER
 #pragma warning(push)
 #pragma warning(disable : 4554)
+#pragma warning(disable : 4324)
```

Comment:
Put it into eigen_common_wrapper.h if it is only for eigen

Reply:
Isn't this the same as C4554 on top ? That seems to be eigen related as well. So you propose moving both ?

Reply:
That would be better if you can move both of them, I think.

Reply:
You may do it like:
```c++
#if defined(__GNUC__)

#elif defined(...)

#endif
```

Instead of 

```c++
#if defined(...)
#endif

#if defined(...)
#endif

```